openlibrary/plugins/upstream/utils.py

```diff
@@ -290,7 +290,9 @@ def setvalue(data, k, v):
             k, k2 = k.split(separator, 1)
             setvalue(data.setdefault(k, {}), k2, v)
         else:
-            data[k] = v
+            # Don't overwrite if the key already exists
```

Comment:
XXX May possible affect other forms

Reply:
Yeah I'm a little nervous about this one ; it shouldn't happent often, but this is deep in the form processing core of OL!